Library/Homebrew/cmd/untap.rb

```diff
@@ -16,18 +17,33 @@ def untap_args
 
         Remove a tapped formula repository.
       EOS
-
+      switch "-f", "--force",
+             description: "Untap even if have already installed formulae from this tap."
       min_named 1
     end
   end
 
   def untap
     args = untap_args.parse
 
+    installed_formulae = Formula.installed
+
     args.named.each do |tapname|
       tap = Tap.fetch(tapname)
       odie "Untapping #{tap} is not allowed" if tap.core_tap?
 
+      installed_tap_formulae = installed_formulae.select do |formula|
+        formula.tap == tap
+      end
+
+      unless installed_tap_formulae.length.zero?
+        if args.force?
+          opoo "Untapping #{tap}, but there are formulae installed from it"
+        else
+          odie "Untapping #{tap} cannot be done because of installed formulae from it"
```

Comment:
```suggestion
          odie <<~EOS
            Refusing to untap #{tap} because it contains the following installed formulae or casks:
            #{(installed_tap_formulae + installed_tap_casks.map(&:token)).join("\n")}
          EOS
```